File: app/src/test/java/org/jellyfin/androidtv/ui/playback/overlay/CustomSeekProviderTest.java
package org.jellyfin.androidtv.ui.playback.overlay;

import org.junit.Test;

import static org.junit.Assert.assertArrayEquals;
import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertTrue;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

public class CustomSeekProviderTest {

    @Test
    public void testGetSeekPositions_withSimpleDuration() {
        VideoPlayerAdapter videoPlayerAdapter = mock(VideoPlayerAdapter.class);
        when(videoPlayerAdapter.canSeek()).thenReturn(true);
        when(videoPlayerAdapter.getDuration()).thenReturn(90_000L);

        CustomSeekProvider customSeekProvider = new CustomSeekProvider(videoPlayerAdapter);

        long[] expected = new long[]{0L, 30_000L, 60_000L, 90_000L};
        long[] actual = customSeekProvider.getSeekPositions();

        assertArrayEquals(expected, actual);
    }

    @Test
    public void testGetSeekPositions_withOddDuration() {
        VideoPlayerAdapter videoPlayerAdapter = mock(VideoPlayerAdapter.class);
        when(videoPlayerAdapter.canSeek()).thenReturn(true);
        when(videoPlayerAdapter.getDuration()).thenReturn(130_000L);

        CustomSeekProvider customSeekProvider = new CustomSeekProvider(videoPlayerAdapter);

        long[] expected = new long[]{0L, 30_000, 60_000, 90_000, 120_000, 130_000};
        long[] actual = customSeekProvider.getSeekPositions();

        assertArrayEquals(expected, actual);
    }

    @Test
    public void testGetSeekPositions_withSeekDisabled() {
        VideoPlayerAdapter videoPlayerAdapter = mock(VideoPlayerAdapter.class);
        when(videoPlayerAdapter.canSeek()).thenReturn(false);

        CustomSeekProvider customSeekProvider = new CustomSeekProvider(videoPlayerAdapter);

        long[] actual = customSeekProvider.getSeekPositions();

        assertEquals(0, actual.length);
    }
}

File: app/src/test/java/org/jellyfin/androidtv/ui/playback/overlay/CustomSeekProviderTest.kt
package org.jellyfin.androidtv.ui.playback.overlay

import org.junit.Assert
import org.junit.Test
import org.mockito.Mockito

class CustomSeekProviderTest {
	@Test
	fun testGetSeekPositions_withSimpleDuration() {
		val videoPlayerAdapter = Mockito.mock(
			VideoPlayerAdapter::class.java
		)
		Mockito.`when`(videoPlayerAdapter.canSeek()).thenReturn(true)
		Mockito.`when`(videoPlayerAdapter.duration).thenReturn(90000L)
		val customSeekProvider = CustomSeekProvider(videoPlayerAdapter)
		val expected = longArrayOf(0L, 30000L, 60000L, 90000L)
		val actual = customSeekProvider.seekPositions
		Assert.assertArrayEquals(expected, actual)
	}

	@Test
	fun testGetSeekPositions_withOddDuration() {
		val videoPlayerAdapter = Mockito.mock(
			VideoPlayerAdapter::class.java
		)
		Mockito.`when`(videoPlayerAdapter.canSeek()).thenReturn(true)
		Mockito.`when`(videoPlayerAdapter.duration).thenReturn(130000L)
		val customSeekProvider = CustomSeekProvider(videoPlayerAdapter)
		val expected = longArrayOf(0L, 30000, 60000, 90000, 120000, 130000)
		val actual = customSeekProvider.seekPositions
		Assert.assertArrayEquals(expected, actual)
	}

	@Test
	fun testGetSeekPositions_withSeekDisabled() {
		val videoPlayerAdapter = Mockito.mock(
			VideoPlayerAdapter::class.java
		)
		Mockito.`when`(videoPlayerAdapter.canSeek()).thenReturn(false)
		val customSeekProvider = CustomSeekProvider(videoPlayerAdapter)
		val actual = customSeekProvider.seekPositions
		Assert.assertEquals(0, actual.size.toLong())
	}
}
